.github/workflows/build-check.yml

```
@@ -0,0 +1,28 @@
+name: Build check
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  push:
+    branches:
+      - main
+
+jobs:
+  compile:
+    strategy:
+      matrix:
+        os: [
+          ubuntu-latest,
+#          windows-latest,
```

```suggestion
          windows-latest
```
Windows definitely needs to run because of build constraints. Mac runs in the same plugin with Linux so yeah we can skip it.

---

This is fine. We don't run any windows tests right now so including this means this PR will take longer. I will deal with windows.